.github/workflows/test-query-engine.yml

```diff
@@ -20,115 +20,137 @@ concurrency:
 
 jobs:
   postgres:
-    if: github.event_name == 'push' || (github.event_name == 'pull_request' && matrix.run_on_pr)
     strategy:
       fail-fast: false
       matrix:
         database:
           - name: 'postgres16'
             version: '16'
-            run_on_pr: true
+    name: ${{ matrix.database.name }}
+    uses: ./.github/workflows/test-query-engine-template.yml
+    with:
+      name: ${{ matrix.database.name }}
+      version: ${{ matrix.database.version }}
+      connector: 'postgres'
+      single_threaded: true
+
+  postgres-push:
+    if: github.event_name == 'push'
+    strategy:
+      fail-fast: false
+      matrix:
+        database:
           - name: 'postgres15'
             version: '15'
-            run_on_pr: false
           - name: 'postgres14'
             version: '14'
-            run_on_pr: false
           - name: 'postgres13'
             version: '13'
-            run_on_pr: false
           - name: 'postgres12'
             version: '12'
-            run_on_pr: false
           - name: 'postgres11'
             version: '11'
-            run_on_pr: false
           - name: 'postgres10'
             version: '10'
-            run_on_pr: false
           - name: 'postgres9'
             version: '9'
-            run_on_pr: false
+    name: ${{ matrix.database.name }}
```

Comment:
I think they show up in the UI like this:
<img width="284" alt="Screenshot 2025-04-14 at 10 39 09" src="https://github.com/user-attachments/assets/8ff8288c-f329-4ef4-8269-22c23165db50" />

Maybe better to call them here:
```suggestion
    name: Postgres Old(v${{ matrix.database.version}})
```

Reply:
Nice catch. Fixed it for all db flavors.